### PR TITLE
Improvements to vertex/silicon wrapper

### PR DESCRIPTION
**Pull request by armin-ilg**
BEGINRELEASENOTES
- No more warnings in silicon wrapper
- Improvements in vertex builder printouts
- Adding all materials of beam pipe also to material_plots_2D.py, as without having the beam pipe enabled, the vertex material budget estimation will fail.
- Changed paths to .stl files in vertex to use https://fccsw.web.cern.ch/fccsw/filesForSimDigiReco/IDEA/IDEA_o1_v03/STL_FILES/ (still commented out due to overlaps)
ENDRELEASENOTES

@BrieucF: As you requested I fixed the warnings in the silicon wrapper.

P.s:
I don't know how to squash all the commits, sorry that this is such a mess

### CODE REVIEW

**Comment by andresailer** [comment]
Can you stop touching your branch for 5 minutes?

File: utils/material_plots_2D.py [hunk 1]
```diff
@@ -36,8 +36,9 @@ def main():
 
         entry_x0, entry_lambda, entry_depth = 0.0, 0.0, 0.0
         for i in range(nMat):
-            if entry.material.at(i) == "Air": continue
-
+            # Ignore materials used in beam pipe as vertex material budget does not work without beam pipe. You can add more materials to ignore here
```
Comment (andresailer) [inline utils/material_plots_2D.py:39]: This makes this script very non-generic and confusing to use.
Maybe you add an option to ignore a list material names?

Comment (andresailer) [reply utils/material_plots_2D.py]: Yes! Thanks!

File: utils/material_plots.py [hunk 2]
```diff
@@ -30,16 +32,14 @@ def main():
         for i in range(nMat):
             material = entry.material.at(i)
 
-            # If you need to replace some string in the material, add that here
-            material = material.replace("66D","")
-            material = material.replace("Vtx","")
+            # Removing substrings from materials
+            for substring in args.removeMatsSubstrings:
+                material = material.replace(substring,"")
```
Comment (BrieucF) [inline utils/material_plots.py:37]: Isn't there an indentation problem here?

Comment (armin-ilg) [reply utils/material_plots.py:37]: I don't see a problem, the indentation is the same as everywhere else (four spaces) and the "Removing substrings from materials" and "Ignore certain materials if specified" functionalities should be at the same level

Comment (BrieucF) [reply utils/material_plots.py:37]: Indeed, I mis-read, sorry

**Comment by BrieucF** [comment]
Shall we merge this before or after https://github.com/key4hep/k4geo/pull/392?

**Comment by tmadlener** [comment]
There will most likely be a conflict either way. So, I don't have any strong preferences.

**Pull request merged**